### ISSUE TITLE
archlinux - dummyqbs_drv.so not loading (undefined symbol: fbScreenInit)

### ISSUE THREAD

**Issue by ptitdoc** [QubesOS/qubes-issues#3093]
#### Qubes OS version (e.g., `R3.2`):
R3.2

#### Affected TemplateVMs (e.g., `fedora-23`, if applicable):
archlinux

---

### Expected behavior:
xorg starts properly and the GUI agent works properly.

### Actual behavior:
the AppVM starts properly with qrexec working, the systemd qubes gui daemon starts properly, but no GUI application can be started.

### Steps to reproduce the behavior:
Install qubes-vm-gui-3.2.17-7-x86_64.pkg.tar.xz in archlinux.

### General notes:
- /home/user/.xsession-errors shows Fatal error "no screens found"
- /home/user/.local/share/xorg/Xorg.0.log shows the following message:
```
[     6.358] (II) LoadModule: "dummyqbs"
[     6.360] (II) Loading /usr/lib/xorg/modules/drivers/dummyqbs_drv.so
[     6.360] (EE) Failed to load /usr/lib/xorg/modules/drivers/dummyqbs_drv.so: /usr/lib/xorg/modules/drivers/dummyqbs_drv.so: undefined symbol: fbScreenInit
[     6.360] (II) UnloadModule: "dummyqbs"
[     6.360] (II) Unloading dummyqbs
[     6.360] (EE) Failed to load module "dummyqbs" (loader failed, 7)
[     6.360] (II) LoadModule: "qubes"
[     6.360] (II) Loading /usr/lib/xorg/modules/drivers/qubes_drv.so
```
---

**Comment by ptitdoc** [comment]
This can be fixed by forcing Xorg to load libfb.
For instance, by adding to the file `/etc/X11/xorg-qubes.conf.template`:
```
Section "Module"
  Load "libfb.so"
EndSection
```

However, I have no hints why Xorg stopped loading this module automatically.

**Comment by ptitdoc** [comment]
If I check on a system before updating, in Xorg logs, libfb.so is loaded after dummyqbs_drv.so however if I check the .so file (nm -C libfb.so), libfb exports fbScreenInit.

If I check on a system after the update, in Xorg logs, dummyqbs_drv.so is loaded immediatly after glx and fail to load because of undefined symbol: fbScreenInit. If I check nm -C libfb.so, it exports fbScreenInit.

**Comment by ptitdoc** [comment]
Do anyone see problems enforcing libfb.so loading into xorg-qubes.conf.template ? This could fix future issues with other OSes... or not...

**Comment by marmarek** [comment]
Since we don't have diverse X configurations to support there, loading `libfb.so` explicitly is ok.
I assume, you've already tried rebuilding gui-agent (with current Xorg headers and libs), right?

**Comment by ptitdoc** [comment]
I tried to build the gui-agent twice, using the builder. Additionnally, I think that loading a module build with an old Xorg server version in a new Xorg server will raise version mismatch errors (there is however an option in xorg config to accept version mismatch errors).

Anyway, I checked 1 hours without fiding the potential cause. This also could be an Xorg driver loader bug.
One strange thing however is that before this bug, the modules where loaded in the wrong order, without raising any undefined symbol errors.
